drivers/wiegand/wiegand.c

```diff
@@ -0,0 +1,329 @@
+{
+    return dev->type;
+}
+
+/**
+ * @brief Translates * and # keys to [ENTER] and [ESCAPE] keys.
+ *
+ */
+static char translate_enter_escape_key_press(char originalKeyPress)
+{
+    switch (originalKeyPress) {
+        case 0x0b:          // 11 or * key
+            return 0x0d;    // 13 or ASCII ENTER
+
+        case 0x0a:          // 10 or # key
+            return 0x1b;    // 27 or ASCII ESCAPE
+
+        default:
+            return originalKeyPress;
+    }
+}
+
+/**
+ * @brief Disables interrupt handling for D0 and D1 lines.
+ *
+ */
+static inline void wiegand_disable_int(wiegand_t *dev)
+{
+    gpio_irq_disable(dev->conn->d0);
+    gpio_irq_disable(dev->conn->d1);
+}
+
+/**
+ * @brief Enables interrupt handling for D0 and D1 lines.
+ *
+ */
+static inline void wiegand_enable_int(wiegand_t *dev)
+{
+    gpio_irq_enable(dev->conn->d0);
+    gpio_irq_enable(dev->conn->d1);
+}
+
+/**
+ * @brief Returns the card identification.
+ *
+ */
+static unsigned long wiegand_get_card_id(volatile unsigned long *codehigh,
+                                         volatile unsigned long *codelow,
+                                         char bitlength)
+{
+
+    if (bitlength == 26) { // EM tag
+        return (*codelow & 0x1FFFFFE) >> 1;
+    }
+
+    if (bitlength == 34) {              // Mifare
+        *codehigh = *codehigh & 0x03;   // only need the 2 LSB of the codehigh
+        *codehigh <<= 30;               // shift 2 LSB to MSB
+        *codelow >>= 1;
+        return *codehigh | *codelow;
+    }
+    return *codelow; // EM tag or Mifare without parity bits
+}
+
+/**
+ * @brief Process the temporary Wiegand buffer and decodes the data.
+ *
+ */
+static bool wiegand_do_conversion(wiegand_t *dev)
+{
+    uint64_t sysTick;
+
+    sysTick = xtimer_now_usec64();
+    // Here we check if we don't have recent activity
+    // TODO: This code smells. Please fix.
+    if (
+        sysTick - dev->wg_last_wiegand > 25000 &&
+        dev->wg_bit_count != 0 &&
+        dev->wg_bit_count == dev->last_bit_count) {
+        dev->ready = 1;
+    }
+    dev->wg_last_wiegand = sysTick;
+    dev->last_bit_count = dev->wg_bit_count;
+
+    if (dev->ready) { // If thereis no recent activity
+        // bitCount for keypress=4 or 8, Wiegand 26=24 or 26,
+        // Wiegand 34=32 or 34
+        if (
+            (dev->wg_bit_count == 24) || (dev->wg_bit_count == 26) ||
+            (dev->wg_bit_count == 32) ||
+            (dev->wg_bit_count == 34) || (dev->wg_bit_count == 8) ||
+            (dev->wg_bit_count == 4)) {
+
+            // Shift right 1 bit to get back the real value - interrupt
+            // done 1 left shift in advance
+            dev->wg_card_temp >>= 1;
+
+            // Bit count more than 32 bits, shift high bits right to make
+            // adjustment
+            if (dev->wg_bit_count > 32) {
+                dev->wg_card_temp_high >>= 1;
+            }
+
+            if (dev->wg_bit_count == 8) { // keypress wiegand with integrity
```

Comment:
I think this would look cleaner if you did a 

```C
switch (dev->wg_bit_count) {
case 4:
    …
case 8:
    …
```

Reply:
Done. I cleaned up a little bit the whole section of the code.